src/react-native-animated-charts/src/charts/linear/ChartPathProvider.tsx

```diff
@@ -0,0 +1,206 @@
+
+        const scaleX = scaleLinear()
+          .domain([smallestX, greatestX])
+          .range([0, width]);
+
+        const scaleY = scaleLinear()
+          .domain(yRange ?? [smallestY, greatestY])
+          .range([height, 0]);
+
+        return {
+          scaleY,
+          scaleX,
+        };
+      },
+      []
+    );
+
+    const createPath = useCallback(
+      ({ data, width, height, yRange }: CallbackType): PathData => {
+        const { scaleY, scaleX } = getScales({
+          data,
+          width,
+          height,
+          yRange,
+        });
+
+        if (!data.points.length) {
+          return {
+            path: '',
+            parsed: null,
+            points: [],
+            data: [],
+          };
+        }
+
+        const points: Point[] = [];
+
+        const { greatestY, smallestY } = findYExtremes(data.points) as {
+          greatestY: Point;
+          smallestY: Point;
+        };
+        const smallestX = data.points[0];
+        const greatestX = data.points[data.points.length - 1];
+
+        for (let i = 0; i < data.points.length; i++) {
+          points.push({
+            x: scaleX(data.points[i].x),
+            y: scaleY(data.points[i].y),
+          });
+        }
+
+        const path = shape
+          .line()
+          .x((item: Point) => scaleX(item.x))
+          .y((item: Point) => scaleY(item.y))
+          .curve(getCurveType(data.curve!))(data.points) as string;
+
+        const parsed = redash.parse(path);
+
+        return {
+          path,
+          parsed,
+          points,
+          data: data.points,
+          smallestX,
+          smallestY,
+          greatestX,
+          greatestY,
+        };
+      },
+      []
+    );
+
+    const initialized = useRef(false);
+
+    const initialPath = useMemo(
+      () =>
+        data.points.length ? createPath({ data, width, height, yRange }) : null,
+      []
+    );
+
+    const [paths, setPaths] = useState<[PathData | null, PathData | null]>(
+      () => [initialPath, initialPath]
+    );
+
+    const previousPath = paths[0];
+    const currentPath = paths[1];
+
+    useEffect(() => {
+      if (initialized.current) {
+        setPaths(([_, curr]) => [
+          curr,
+          data.points.length
+            ? createPath({ data, width, height, yRange })
+            : null,
+        ]);
+      } else {
+        initialized.current = true;
+      }
+    }, [data.points, data.curve, width, height]);
+
+    const value = useMemo(() => {
+      const ctx = {
+        progress,
+        dotScale,
+        originalX,
+        originalY,
+        pathOpacity,
+        state,
+        positionX,
+        positionY,
+        isActive,
+        data,
+        width,
+        height,
+        previousPath,
+        currentPath,
+      };
+
+      if (currentPath) {
+        const { smallestX, smallestY, greatestX, greatestY } = currentPath;
+        Object.assign(ctx, { smallestX, smallestY, greatestX, greatestY });
```

Comment:
why not 
```
	return {
		...ctx,
		smallestX,
		smallestY,
		greatestX,
		greatestY 
	}
		
	}
```
?

Reply:
I can use this too. My version is just a bit more "optimized" since we don't create yet another object where we copy all the stuff from ctx and third object. Instead, we just mutate ctx with new stuff. But for readability I can use your version i guess.